src/pages/login.svelte

```diff
@@ -12,6 +15,52 @@
     import MediaQuery from '~/components/utils/media-query.svelte'
     import Features from '~/components/elements/features.svelte'
     import UnicoveAnimated from '~/components/elements/unicove-animated.svelte'
+
+    const whalesplainerUrl = import.meta.env.SNOWPACK_PUBLIC_WHALESPLAINER_URL
+
+    let creatingAccount = false
+
+    function handleCreateAccount(event: Event) {
+        event.preventDefault()
+
+        createAccount().catch((error) => {
+            addToast({
+                title: 'Account not created!',
+                message: error.message,
+                timeout: 10000,
+            })
+        })
+    }
+
+    function createAccount() {
+        return new Promise((resolve, reject) => {
```

Comment:
You should never need to create a new promise manually unless you are bridging a callback style API to a promise flow. Why did you remove the async function?

Reply:
Because I needed a way to catch the error that happened in this function and the sync function wasn't catching the errors that  were occurring in the account creator catch function. Let me try something else.

Reply:
@jnordberg I turned it back into an async function, let me know what you think.

Reply:
Looking better, now errors are caught. If you want to make it more clean without the try/catch inside the async function you can move the side effect that sets `creatingAccount` out to the `handleCreateAccount` function and treat the async `createAccount` as a pure function. Something like

```ts
function handleCreateAccount() {
    creatingAccount = true
    createAccount()
        .then(() => showToast('yay!'))
        .catch((error) => showError(error))
        .finally(() => {
            creatingAccount = false
        })
}
```

That will also have the benefit of creatingAccount not getting stuck to true if there ever is an unexpected error outside of your try catch block.

Reply:
That is much cleaner! Thanks!